packages/react/src/components/select/Select.stories.tsx

```diff
@@ -152,10 +162,117 @@ export const WithGroups = () => {
   ];
 
   return (
-    <Select groups={groups} label="Label" placeholder="Choose one" icon={<IconLocation />}>
+    <Select groups={groups} icon={<IconLocation />} onChange={dummyOnChange} texts={defaultTexts}>
       <optgroup label="Group label">
         <option value="label">Text</option>
       </optgroup>
     </Select>
   );
 };
+
+export const WithControls = () => {
+  const initialGroups = [
+    {
+      label: 'Healthy choices',
+      options: generateOptionLabels(4),
+    },
+    {
+      label: 'More healthy choices',
+      options: generateOptionLabels(4),
+    },
+  ];
+
+  const onChange: SelectProps['onChange'] = useCallback(() => {
+    // not needed
+  }, []);
+
+  const [props, updateProps] = useState<SelectProps>({
+    groups: propsToGroups({ groups: initialGroups }),
+    onChange,
+    required: true,
+    disabled: false,
+    open: false,
+    invalid: false,
+    texts: { ...defaultTexts, label: 'Controlled select' },
+  });
+
+  const resetSelections = () => {
+    updateProps({ ...props, groups: clearAllSelectedOptions(props.groups as Group[]), open: false });
+  };
+
+  const toggleDisable = () => {
+    const current = props.disabled;
```

Comment:
Should these be written like `const { disabled } = props;` and then `updateProps({ ...props, disabled: !disabled });`. Though dunno which is better 🤷

Reply:
Your suggested way is cleared! Good point! Fixed.